From 1b3c2bf1aa20eb17cbafd54e85e7afe806a91aaf Mon Sep 17 00:00:00 2001
From: Jackie Huang <jackie.huang@windriver.com>
Date: Wed, 12 Jul 2023 10:32:54 +0800
Subject: [PATCH 2/2] puppet-manifests: add arm64 support for etcd

Add environmnet ETCD_UNSUPPORTED_ARCH=arm64 for etcd
so it will workaround the unsupported warning.

Test Plan:
PASS: build-pkgs on x86-64 host
PASS: build-image on x86-64 host
PASS: build-pkgs on arm64 host
PASS: build-image on arm64 host
PASS: install AIO-SX on x86-64 targets
PASS: install AIO-SX on arm64 targets

Story: 2010739
Task: 48284

Signed-off-by: Jackie Huang <jackie.huang@windriver.com>
---
 puppet-manifests/debian/deb_folder/rules | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/puppet-manifests/debian/deb_folder/rules b/puppet-manifests/debian/deb_folder/rules
index f5ba9a01..081e869c 100755
--- a/puppet-manifests/debian/deb_folder/rules
+++ b/puppet-manifests/debian/deb_folder/rules
@@ -14,6 +14,8 @@ override_dh_usrlocal:
 override_dh_install:
 ifeq ($(DEB_HOST_ARCH),arm64)
 	sed -i "s/amd64/arm64/" $(CURDIR)/bin/puppet-update-grub-env.py
+	sed -i '/EnvironmentFile/a Environment="ETCD_UNSUPPORTED_ARCH=arm64"' \
+		$(CURDIR)/modules/platform/templates/etcd-override.conf.erb
 endif
 	$(MAKE) install hiera_v5=true ignore_puppet_warnings=true \
 		BINDIR=$(CURDIR)/debian/tmp/usr/local/bin \
-- 
2.30.2

